File: uva_10394.cpp
# include <vector>
# include <iostream>
#define MAX 20000000
using namespace std;
int main(void){
    vector<bool> prime(20000000, true);
    vector<int> twin;
    for (int i=2; i<MAX; ++i){
        if (prime[i])
            for (int j=2*i; j<MAX; j+=i)
                prime[j]=false;
        if (prime[i-2]&&prime[i])
            twin.push_back(i-2);
    }
    int S;
    while(cin >> S)
        cout << "(" << twin[S+1] << ", " << twin[S+1]+2 << ")\n";
    return 0;
}

File: uva_10394.py
MAX=20000000
prime=[True for _ in range(MAX)]
twin=[]
for i in range(2, MAX):
    if prime[i]:
        j=2*i
        while j<MAX:
            prime[j]=False
            j+=i
    if prime[i-2] and prime[i]:
        twin.append(i-2)
while True:
    try:
        S=int(input())
    except:
        break
    print("({}, {})".format(twin[S+1], twin[S+1]+2))
